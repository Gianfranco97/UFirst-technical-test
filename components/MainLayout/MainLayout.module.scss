.layout {
  height: 100vh;
}

.header {
  h1 {
    color: white;
  }
}

.footer,
.header {
  text-align: center;
}
