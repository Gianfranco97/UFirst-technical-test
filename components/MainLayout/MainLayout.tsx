import { Layout } from 'antd';
import styles from "./MainLayout.module.scss"

type DashboardLayoutProps = {
  children: React.ReactNode,
};

export default function MainLayout({ children }: DashboardLayoutProps) {
  return (
    <Layout className={styles.layout}>
      <Layout.Header className={styles.header}><h1>Technical test - UFirst</h1></Layout.Header>

      <Layout>
        <Layout.Content className={styles.main}>{children}</Layout.Content>
      </Layout>

      <Layout.Footer className={styles.footer}>Â© 2022 - Gianfranco Manganiello</Layout.Footer>
    </Layout>
  )
}