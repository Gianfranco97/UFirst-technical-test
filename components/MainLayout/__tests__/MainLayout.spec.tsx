import { expect } from '@jest/globals';
import { render } from '@testing-library/react'
import MainLayout from '../MainLayout';

describe('MainLayout', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MainLayout><h1>Hello World</h1></MainLayout>
    )

    expect(container).toMatchSnapshot()
  })
});