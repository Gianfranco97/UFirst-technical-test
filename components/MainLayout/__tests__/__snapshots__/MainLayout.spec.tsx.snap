// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders correctly 1`] = `
<div>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header header"
    >
      <h1>
        Technical test - UFirst
      </h1>
    </header>
    <section
      class="ant-layout"
    >
      <main
        class="ant-layout-content main"
      >
        <h1>
          Hello World
        </h1>
      </main>
    </section>
    <footer
      class="ant-layout-footer footer"
    >
      Â© 2022 - Gianfranco Manganiello
    </footer>
  </section>
</div>
`;
